aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -19,6 +19,298 @@ The model classes you create are known as POCO classes (from **P**lain **O**ld *
+
+In **Solution Explorer**, right-click the *Controllers* folder **> Add > New Scaffolded Item**.
+
+![view of above step](adding-model/_static/add_controller21.png)
+
+In the **Add Scaffold** dialog, select **MVC Controller with views, using Entity Framework > Add**.
+
+![Add Scaffold dialog](adding-model/_static/add_scaffold21.png)
+
+Complete the **Add Controller** dialog:
+
+* **Model class:** *Movie (MvcMovie.Models)*
+* **Data context class:** Select the **+** icon and change the generated name from MvcMovie.**Models**.MvcMovieContext to MvcMovie.**Data**.MvcMovieContext. [This change](https://developercommunity.visualstudio.com/content/problem/652166/aspnet-core-ef-scaffolder-uses-incorrect-namespace.html) is not required. It creates the database context class with the correct namespace.
+
+![Add Data context](adding-model/_static/dc3.png)
+
+* **Views:** Keep the default of each option checked
+* **Controller name:** Keep the default *MoviesController*
+* Select **Add**
+
+Visual Studio creates:
+
+* An Entity Framework Core [database context class](xref:data/ef-mvc/intro#create-the-database-context) (*Data/MvcMovieContext.cs*)
+* A movies controller (*Controllers/MoviesController.cs*)
+* Razor view files for Create, Delete, Details, Edit, and Index pages (*Views/Movies/\*.cshtml*)
+
+The automatic creation of the database context and [CRUD](https://wikipedia.org/wiki/Create,_read,_update_and_delete) (create, read, update, and delete) action methods and views is known as *scaffolding*.
+
+### [Visual Studio Code](#tab/visual-studio-code) 
+
+* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+* On Linux, export the scaffold tool path:
+
+  ```console
+    export PATH=$HOME/.dotnet/tools:$PATH
+  ```
+
+* Run the following command:
+
+  ```console
+     dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
+  ```
+
+[!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+
+### [Visual Studio for Mac](#tab/visual-studio-mac)
+
+* Open a command window in the project directory (The directory that contains the *Program.cs*, *Startup.cs*, and *.csproj* files).
+
+* Run the following command:
+
+  ```console
+     dotnet aspnet-codegenerator controller -name MoviesController -m Movie -dc MvcMovieContext --relativeFolderPath Controllers --useDefaultLayout --referenceScriptLibraries
+  ```
+
+[!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+---
+
+<!-- End of tabs                  -->
+
+If you run the app and click on the **Mvc Movie** link, you get an error similar to the following:
+
+``` error
+An unhandled exception occurred while processing the request.
+
+SqlException: Cannot open database "MvcMovieContext-<GUID removed>" requested by the login. The login failed.
```

Comment:
It's a different exception with SQLite -- "SqliteException: SQLite Error 1: 'no such table: Movie'."